scripts/leapp_script.py

```diff
@@ -97,11 +101,21 @@ def get_rhel_version():
 
 
 def is_non_eligible_releases(release):
-    print("Exit if not RHEL 7.9 or 8.4")
-    eligible_releases = ["7.9", "8.4"]
+    """Check if the release is eligible for upgrade or preupgrade."""
+    not_eligable_condition = False
     major_version, minor = release.split(".") if release is not None else (None, None)
-    version_str = major_version + "." + minor
-    return release is None or version_str not in eligible_releases
+    if IS_UPGRADE:
+        print("Exit if not RHEL 7.9 or 8.4")
+        eligible_releases = ["7.9", "8.4"]
+        version_str = major_version + "." + minor
+        not_eligable_condition = version_str not in eligible_releases
+    elif IS_PREUPGRADE:
+        print("Exit if not RHEL 7 or RHEL 8 ...")
+        major_version, _ = release.split(".") if release is not None else (None, None)
+        not_eligable_condition = major_version not in ["7", "8"]
+        print(not_eligable_condition)
+
+    return release is None or not_eligable_condition
```

Comment:
Hmm :thinking: this is nice, why do we check specific minor version for upgrade (=we would have to update it) and not preupgrade? Isn't the leapp behavior same in both cases? I mean isn't there inhibitor in report if it's not supported path? We should verify this, I think checking for major 7 and 8 should be enough (and in future tasks backend will take care of pre-filtering of the eligible systems)

Reply:
I find [this](https://github.com/oamg/leapp-repository/blob/353cd03d5339a6f3905f8bc4f067e0758f6e1d78/repos/system_upgrade/common/actors/checkosrelease/libraries/checkosrelease.py#L25) so it looks like we don't even need this check, but we can keep it at least for RHEL 7 and 8 majors.